qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1090,4 +1100,64 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 		}
 		return resolvedJavaType;
 	}
-}
+
+	private void validateSwitchWhenPart(JavaMemberInfo member, Section ownerSection, String projectUri,
+			ResolvingJavaTypeContext resolvingJavaTypeContext, PropertyPart propertyPart, MethodPart methodPart,
+			List<Diagnostic> diagnostics) {
+		String javaTypeParent = member.getJavaElementType();
+		String partName = "";
+		if (methodPart != null) {
+			partName = methodPart.toString();
+		} else if (propertyPart != null) {
+			partName = propertyPart.toString();
+		}
+		List<Node> children = ownerSection.getChildren();
+		// Iterate through children and ensure the type of the #case or #is parts are
+		// the same as the parent section
+		for (Node node : children) {
+			if (node.getKind() == NodeKind.Section) {
+				List<Parameter> value = new ArrayList<Parameter>();
+				if (ownerSection.getSectionKind() == SectionKind.SWITCH) {
+					value = ((CaseSection) node).getParameters();
+				} else if (ownerSection.getSectionKind() == SectionKind.WHEN) {
+					value = ((IsSection) node).getParameters();
+				}
+				boolean isKeywordIn = true;

```

Comment:
You can simplify the in keyword test by checking if the first parameter is "in" and if there are parameters size > 1.

Here the code I suggest you:

```java
List<Parameter> parameters = childSection.getParameters();
for (int i = 0; i < parameters.size(); i++) {
	Parameter parameter = parameters.get(i);
	String parameterName = parameter.getName();
	if (i == 0 && "in".equals(parameterName) && parameters.size() > 1) {
		// In the of 'in', the first is allowed as a keyword, second must be checked
		// Ex: {#is in in BROKEN}
		continue;
	}
	String javaTypeChild = LiteralSupport.getLiteralJavaType(parameter.getValue());
	if (javaTypeChild == null) {
		// Case of Enum since it is not a literal java type
		// {#when machine.status} --> machine.status is has values ON and OFF
		// {#is ON} --> valid enum value
		// {#is in BAD_ENUM_VAUE} --> Error: unexpected enum value
		// {/when}
		ResolvedJavaTypeInfo javaTypeChildInfo = resolveJavaType(javaTypeParent, projectUri,
				resolvingJavaTypeContext);
		if (javaCache.findMember(javaTypeChildInfo, parameterName, projectUri) == null) {
			Range range = QutePositionUtility.createRange(parameter);
			Diagnostic diagnostic = createDiagnostic(range, DiagnosticSeverity.Error,
					QuteErrorCode.UnexpectedEnumValue, parameterName, partName, javaTypeParent);
			diagnostics.add(diagnostic);
		}
	} else if (!javaTypeParent.equals(javaTypeChild)) {
		// Case where the java type of expression of child is different from parent
		// Ex: {#switch person.name} --> person.name is of type String
		// {#case 'John'} --> valid expression type of String
		// Hey John!
		// {#case 123} --> Error: unexpected type, expected String but was Integer
		// {/switch}
		Range range = QutePositionUtility.createRange(parameter);
		Diagnostic diagnostic = createDiagnostic(range, DiagnosticSeverity.Error,
				QuteErrorCode.UnexpectedMemberType, javaTypeChild, partName, javaTypeParent);
		diagnostics.add(diagnostic);
	}
}

```